testing_and_setup/compass/ocean/global_ocean/scripts/create_E3SM_coupling_files.py

```diff
@@ -307,195 +334,158 @@ def mapping_analysis(config, mesh_name, date_string, ice_shelf_cavities):
-                           + short + '.' + date_string + '.nc',
-                        '--ignore_unmapped']
-                run_command(args)
-
-                # Atmosphere to ocean
-                args = ['mpirun', '-n', nprocs, 'ESMF_RegridWeightGen',
-                        '--method', method,
-                        '--source', atm_scrip_file,
-                        '--destination', ocn_scrip_file,
-                        '--weight', 'map_' + atm_scrip_tag + '_TO_' + mesh_name + '.mask_'
-                           + short + '.' + date_string + '.nc',
-                        '--ignore_unmapped']
-                run_command(args)
-
-        except OSError:
-            print('mapping_CORE_Gcase must be run on one compute node')
 # }}}
 
 
-def domain_CORE_Gcase(config, mesh_name, date_string, ice_shelf_cavities):
-# {{{
-    # obtain configuration settings
-    domain_exe = config.get('main', 'domain_exe')
-    atm_scrip_tag = config.get('mapping_CORE_Gcase', 'atm_scrip_tag')
-    if ice_shelf_cavities:
-        nomaskStr='.nomask'
-    else:
-        nomaskStr=''
-
-    # make links
-    make_link(domain_exe, 'domain_exe')
-    mapping_file = 'map_' + mesh_name + nomaskStr + '_TO_' + \
-        atm_scrip_tag + '_aave.' + date_string + '.nc'
-    make_link('../mapping_CORE_Gcase/' + mapping_file, mapping_file)
+def domain_CORE_Gcase(config):  # {{{
+    make_domain_files(config, 'CORE_Gcase')
+# }}}
 
-    # execute commands
-    args = ['./domain_exe', '-m', mapping_file, '-o', mesh_name, '-l', 'T62']
-    run_command(args)
 
-    # make links in output directories
-    files = glob.glob('domain*.nc')
-    os.chdir('../assembled_files_for_upload/inputdata/share/domains')
-    for file in files:
-        make_link('../../../../domain/' + file, './' + file)
+def domain_JRA_Gcase(config):  # {{{
+    make_domain_files(config, 'JRA_Gcase')
 # }}}
 
 
-def domain_JRA_Gcase(config, mesh_name, date_string, ice_shelf_cavities):
-# {{{
-    # obtain configuration settings
-    domain_exe = config.get('main', 'domain_exe')
-    atm_scrip_tag = config.get('mapping_JRA_Gcase', 'atm_scrip_tag')
-    if ice_shelf_cavities:
-        nomaskStr='.nomask'
-    else:
-        nomaskStr=''
+def domain_ne30(config):  # {{{
+    make_domain_files(config, 'ne30')
+# }}}
 
-    # make links
-    make_link(domain_exe, 'domain_exe')
-    mapping_file = 'map_' + mesh_name + nomaskStr + '_TO_' + \
-        atm_scrip_tag + '_aave.' + date_string + '.nc'
-    make_link('../mapping_JRA_Gcase/' + mapping_file, mapping_file)
 
-    # execute commands
-    args = ['./domain_exe', '-m', mapping_file, '-o', mesh_name, '-l', 'T62']
-    run_command(args)
-
-    # make links in output directories
-    files = glob.glob('domain*.nc')
-    os.chdir('../assembled_files_for_upload/inputdata/share/domains')
-    for file in files:
-        make_link('../../../../domain/' + file, './' + file)
-# }}}
+def make_domain_files(config, mapping_suffix):  # {{{
```

Comment:
This function replaces the 3 `domain` functions above.  We could add more arguments in the future if needed (e.g. different atmosphere grids.